python/tank/pipelineconfig.py

```diff
@@ -934,15 +944,23 @@ def execute_core_hook_method_internal(self, hook_name, method_name, parent, **kw
         file_name = "%s.py" % hook_name
         hooks_path = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "..", "hooks"))
         hook_paths = [os.path.join(hooks_path, file_name)]
+
+        # the hook.method display name used when logging the metric and when
+        # identifying the metric to be logged only once in the system.
+        hook_method_display = "%s.%s" % (hook_name, method_name)
         
         # now add a custom hook if that exists.
         hook_folder = self.get_core_hooks_location()        
         hook_path = os.path.join(hook_folder, file_name)
         if os.path.exists(hook_path):
             hook_paths.append(hook_path)
-            if hasattr(parent, 'log_metric'):
-                parent.log_metric("custom hook %s" % (hook_name,))
+            if (hasattr(parent, 'log_metric') and
+                not hook_method_display in self.__logged_custom_hook_metrics and
+                hook_name not in constants.TANK_LOG_METRICS_CUSTOM_HOOK_BLACKLIST):
 
+                # log and remember the name so that we don't log it again
+                parent.log_metric("custom hook %s" % (hook_method_display,))
```

Comment:
I am sure i am missing something obvious here, so again apologies in advance - but could we do

```python
parent.log_metric("custom hook %s" % (hook_method_display,), log_only_once=True)
```

Whenever the metrics singleton gets the `log_only_once=True`, it puts the message into a set for later comparison, making sure that whenever that `log_only_once=True` flag is set, the metrics string passed to the method is only emitted once per singleton session.

If we wanted to control some hooks like this but not all, I guess we could introduce a constant, similar to `TANK_LOG_METRICS_CUSTOM_HOOK_BLACKLIST` and hard code a list. Something like

```
TANK_LOG_METRICS_LOG_ONCE=["cache_location.get_path_cache_path"]

# and then
parent.log_metric(
    "custom hook %s" % (hook_method_display,),
    log_only_once = (hook_method_display in constants.TANK_LOG_METRICS_LOG_ONCE)
    )
```

You have a better understanding of all this - i think if we could avoid tracking metrics state outside the metrics system (e.g. somehow avoid `__logged_custom_hook_metrics` inside the pipelineconfiguation class) that would be great, but if it's doesn't make practical sense, then that's cool too.